phpBB/phpbb/ucp/controller/reset_password.php

```diff
@@ -0,0 +1,443 @@
+<?php
+/**
+*
+* This file is part of the phpBB Forum Software package.
+*
+* @copyright (c) phpBB Limited <https://www.phpbb.com>
+* @license GNU General Public License, version 2 (GPL-2.0)
+*
+* For full copyright and license information, please see
+* the docs/CREDITS.txt file.
+*
+*/
+
+namespace phpbb\ucp\controller;
+
+use phpbb\auth\auth;
+use phpbb\config\config;
+use phpbb\controller\helper;
+use phpbb\db\driver\driver_interface;
+use phpbb\event\dispatcher;
+use phpbb\exception\http_exception;
+use phpbb\language\language;
+use phpbb\log\log_interface;
+use phpbb\passwords\manager;
+use phpbb\request\request_interface;
+use phpbb\template\template;
+use phpbb\user;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+* Handling forgotten passwords via reset password functionality
+*/
+class reset_password
+{
+	/** @var config */
+	protected $config;
+
+	/** @var driver_interface */
+	protected $db;
+
+	/** @var dispatcher */
+	protected $dispatcher;
+
+	/** @var helper */
+	protected $helper;
+
+	/** @var language */
+	protected $language;
+
+	/** @var log_interface */
+	protected $log;
+
+	/** @var manager */
+	protected $passwords_manager;
+
+	/** @var request_interface */
+	protected $request;
+
+	/** @var template */
+	protected $template;
+
+	/** @var user */
+	protected $user;
+
+	/** @var array phpBB DB table names */
+	protected $users_table;
+
+	/** @var string phpBB root path */
+	protected $root_path;
+
+	/** @var string PHP extension */
+	protected $php_ext;
+
+	/**
+	 * Reset password controller constructor.
+	 *
+	 * @param config $config
+	 * @param driver_interface $db
+	 * @param dispatcher $dispatcher
+	 * @param helper $helper
+	 * @param language $language
+	 * @param log_interface $log
+	 * @param manager $passwords_manager
+	 * @param request_interface $request
+	 * @param template $template
+	 * @param user $user
+	 * @param string $users_table
+	 * @param string $root_path
+	 * @param string $php_ext
+	 */
+	public function __construct(config $config, driver_interface $db, dispatcher $dispatcher, helper $helper,
+								language $language, log_interface $log, manager $passwords_manager,
+								request_interface $request, template $template, user $user, string $users_table,
+								string $root_path, string $php_ext)
```

Comment:
will the `string` type declaration work for all php versions supported?

Reply:
That's supported since PHP 7.0 so yes.